### PR TITLE
Search suggestions

### LINKED ISSUE #223: Search suggestions
We need to temporary replicate the `/fields` route from #207 in a new route `/suggestions` to find a way to get search suggestions: each row would for now have a word (and possibly a number of how many documents have that word), same logic we developed in #207.

Later on we'll actually have a different UI for suggestions, this is just about developing/tuning the right query for suggestions.

Max 20 suggestions for now.

See https://towardsdatascience.com/big-fast-nlp-with-elasticsearch-72ffd7ef8f2e?gi=6f4ffea0ca5a

### PR DESCRIPTION
**Pull request by saabi**
Closes #223

### CODE REVIEW

File: src/node_modules/app/components/Search.svelte [hunk 1]
```diff
@@ -17,13 +17,14 @@
 		switch (event.keyCode) {
 			case 40:
 				dispatch('downArrow');
+				event.preventDefault();
```
Comment (mindrones) [inline src/node_modules/app/components/Search.svelte:20]: See https://github.com/nestauk/dapsboard/pull/214#discussion_r555199052

File: src/node_modules/utils/stopwords.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+export const stopWords = [
+    `0`, `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`,
```
Comment (mindrones) [inline src/node_modules/utils/stopwords.js]: I'd substitute 4 blanks with a tab

File: src/node_modules/utils/stopwords.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+export const stopWords = [
```
Comment (mindrones) [inline src/node_modules/utils/stopwords.js:4]: I'd add a comment pointing at where we took this from

File: src/routes/suggestions/index.svelte [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+		// TODO check for error messages
+		let {buckets} = suggestionsResponse.aggregations[fieldName];
+		buckets = buckets.filter(sugg => sugg.key !== searchValue);
+		buckets = buckets.sort((a, b) => b.doc_count - a.doc_count);
+		info.suggestions = buckets.map(sugg => `${sugg.key} (${sugg.doc_count})`);
+		if (info.suggestions.length === 0) {
+			info.suggestions.push('-- no suggestions found --')
+		}
+		fieldCounts = fieldCounts;
+		console.log("countRespoonse", info);
```
Comment (mindrones) [inline src/routes/suggestions/index.svelte:187]: leftover

File: src/routes/suggestions/index.svelte [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+	async function sendSuggestionsRequest (event) {
+		const fieldName = event.detail;
+		const info = fieldCounts.find(fieldInfo => fieldInfo.name === fieldName)
+		if (info.suggestions) {
+			return;
+		}
+		info.suggestions = ['waiting...'];
+		fieldCounts = fieldCounts;
+		const data = computeDetailsQuery(fieldName, searchValue);
```
Comment (mindrones) [inline src/routes/suggestions/index.svelte]: I'd use more specific names for both `data` and `computeDetailsQuery` (we shouldn't have to read a function's code to know what it does)

File: src/routes/suggestions/index.svelte [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+			info.suggestions.push('-- no suggestions found --')
+		}
+		fieldCounts = fieldCounts;
+		console.log("countRespoonse", info);
+	}
+
+	function fieldSelected (event) {
+		userSelection = event.detail;
+		selectedFieldName = computeSelection(fieldCounts);
+	}
```
Comment (mindrones) [inline src/routes/suggestions/index.svelte]: I'd rename `fieldSelected` to `onFieldSelected`

File: src/routes/suggestions/index.svelte [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+		'keywordArray',
+		'textWithKeyword',
+		'textWithKeywordArray'
+	];
+
+	let response;
+	let fieldCounts;
+	let selectedFieldName;
+	let searchWidget;
+
```
Comment (mindrones) [inline src/routes/suggestions/index.svelte]: I'd remove extra lines among vars declarations and sort them alphabetically

**Review by mindrones — COMMENTED** [review]
Left some notes below

File: src/node_modules/app/components/Search.svelte (hunk 1 above)
Comment (saabi) [reply src/node_modules/app/components/Search.svelte:20]: We can't just deduplicate for now since `default:` has different behaviour.

**Review by mindrones — APPROVED** [review]
Great work! 💯

**Pull request merged**